build/image_checks/checkOrphanedImages.js

```diff
@@ -0,0 +1,96 @@
+            path.resolve(common.root_path, `src/javascript/${default_path}`),
+            path.resolve(common.root_path, `src/sass/${default_path}`),
+        ];
+        grunt.log.writeln(`Searching for orphaned images.\nSearching:\nsrc/javascript/${default_path}\nsrc/sass/${default_path}`);
+
+        // Actual test is running from here. Note that this scripts attempt to find the only absolute relative
+        // directory and filename (to detect a broader range of files), this won't allow us to find references
+        // that are made using string concatenations or template strings (mostly in app).
+        const images = fetchFiles(images_directory);
+        let found = flatten(images)
+            .filter(item => !!item &&
+                !ignored_files.some(
+                    ignored => path.resolve(common.root_path, ignored) === item.path
+                ) &&
+                !ignored_image_extensions.some(
+                    ignored => {
+                        const extension = item.filename.match(/\.[\w\d]{1,5}$/i);
+                        return extension != null ? extension[0].substring(1) === ignored : false;
+                    }
+                )
+            );
+
+        grunt.log.ok(`Found ${found.length} files, looking for references.`);
+
+
+        flatten(fetchFiles(source_directory))
+            .filter(
+                item => !!item &&
+                    !ignored_files.some(ignored => path.resolve(common.root_path, ignored) === item.path)
+            )
+            .forEach(item => {
+                // if there is no image to look for, then bail out.
+                if (found.length === 0) {
+                    return false;
+                }
+
+                // Search for the occurrence of images, if found, remove the entry from that array.
+                const file                   = fs.readFileSync(item.full_path, {
+                    encoding: 'utf8',
+                });
+
+                if (!file) {
+                    return false;
+                }
+
+                found = found.filter(dir => file.indexOf(getRelativeImagePath(dir)) === -1);
+                return true;
+            });
+        if (should_delete) {
+            found.forEach(orphan => {
+                grunt.log.writeln(`${orphan.relative_path} removed.`);
+                if (!dry_run) {
+                    fs.unlinkSync(orphan.relative_path);
+                }
+            });
+        } else if (found.length > 0) {
+            grunt.log.error(`Couldn't find references for ${found.length} files.`);
+            grunt.log.writeln('>> Consider find a usage for the following files: ');
+            found.forEach(orphan => grunt.log.writeln(`${orphan.relative_path}`));
+            grunt.log.writeln('='.repeat(80));
+            grunt.log.error(
+                'Some of the files above could be used dynamically.\nIf you are going to delete some, proceed with caution.\n' +
```

Comment:
@easteregg It seems that the result of the task is not reliable, e.g all the images in [this directory](https://github.com/binary-com/binary-static/tree/master/src/images/pages/trade-explanation) are marked as orphaned (at least 40% of the result) but all of them are using.

Reply:
I ignored language images in the past commits, since we are using them dynamically, we cannot detect detect their uses. I try to push a workaround/not-a-fix for it.